### PR TITLE
Add javadoc for Sql Explain Flow -Part II

### PR DESCRIPTION
**Pull request by anthonylouisbsb**
# Pull Request Merge Checklist
           
Issue identifier related to this PR: [434](https://github.com/s1mbi0se/simbiose_knowledge_base/issues/434)

---

## As a DEVELOPER, I confirm that this PR **doesn't contain**:
- [x] Code lines with syntax errors
- [x] Methods and variables names that aren't clear or doesn't mean anything
- [x] Code that is out of the standard (Example: **PEP-8** for Python code)
- [x] Files with **imports** errors or without any utility
- [x] **TODOs** left behind
- [x] Tests that are being skipped
- [x] Methods placed on incorrect files or folders (Example: methods that are local, but should be placed on /utils)
- [x] Functions, methods or API responses that are publicly returning sensitive information

## As a DEVELOPER, I confirm that I:
- [x] Followed the Simbiose's standard for branch and commit naming
- [x] Concluded the task and achieved the goals according to the task planning I defined before starting to code
- [x] Added all necessary database changes to the [SQL Migrations](https://github.com/s1mbi0se/sql-migrations)
- [x] Created tests for every new flow and method on this Pull Request
- [x] Added the necessary DEBUG logs
- [x] Checked my code using the Simbiose's Corner Cases Guide
- [x] Merged the **develop** branch to this PR/branch before sending the PR for review
- [x] Solved all possible merge conflicts before sending the PR for review
- [x] Corrected all the problems and code smells identified by the quality tools
- [x] Manually executed the necessary tests locally to make sure my changes didn't break anything
- [x] All unitary and integration tests ran by the CI system are passing
- [x] Check the performance of my changes or additions and they didn't make the system slower (I profiled the code)
- [x] Updated the necessary documentation (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Made the self-review of the code before sending the PR for review
- [x] Sent this PR for a peer-review
- [x] Made all the corrections appointed during the peer-review
- [x] Confirmed that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block my merge

---  

## As the TEAM MANAGER, I confirm that I:
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked if the purpose of the task was reached with this PR
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if all related PRs were created when other components were affected by this task (e.g. database changes on [SQL Migrations](https://github.com/s1mbi0se/sql-migrations))
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s pattern for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [x] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block the merge

---  

## As the QUALITY RESPONSIBLE, I confirm that I:
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s code standard for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [x] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block the merge

## Important links:
1. [Simbiose Quality Guidelines](https://github.com/s1mbi0se/company/wiki/Quality-Topics#table-of-contents)
2. [Metadata Database Modeling](https://docs.google.com/spreadsheets/d/1Y8_k5xV86kTjgfjbi-PGsJcubUKhWlT2x8ecpnMlyR0/edit?usp=sharing)
3. [Simbiose's Corner Cases Guide](https://www.simbioseventures.com/documentos-e-guias/procedimentos-e-checklists/corner-cases-guide)

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/Session.java [hunk 1]
```diff
@@ -789,11 +789,9 @@ public ResourceEstimateBuilder setPeakMemory(DataSize peakMemory)
         }
 
         /**
-         * Estimate the resource usage for an executed query.
-         * <p>
-         * Takes the estimate of the CPU runtime and memory used for an executed query.
+         * Constructs the object that contains the estimated resource usage of a query.
          *
-         * @return {@link ResourceEstimates} wich shows estimated resource usage for a query.
+         * @return an object that contains the estimated resource usage for a query.
```
Comment (abenaru-zz) [inline presto-main/src/main/java/io/prestosql/Session.java]:
```suggestion
         * @return the estimated resource usage for a query
```
Please refrain from using periods at the end of `@return` tag descriptions.

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 2]
```diff
@@ -209,13 +209,14 @@ public static Identity extractAuthorizedIdentity(Optional<Identity> optionalAuth
     }
 
     /**
-     * Creates a session identity.
+     * Retrieves information about user executing a request and his roles and credentials.
```
Comment (abenaru-zz) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]: Could you rephrase this? I don't think the language used here is very clear. What exactly is this method doing?

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 3]
```diff
@@ -608,7 +611,6 @@ private static void assertRequest(boolean expression, String format, Object... a
      * an WebApplicationException is thrown and handled.
      *
      * @param transactionId a string that represents the transaction id.
-     *
      * @return an Optional TransactionId
```
Comment (abenaru-zz) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:619]: Try to explain the concept rather than the implementation. What is a `TransactionId`? One can already tell this method returns an `Optional<TransactionId>`, so try to add more valuable information about the return value.

**Review by abenaru-zz — CHANGES_REQUESTED** [review]

File: presto-main/src/main/java/io/prestosql/Session.java (hunk 1 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/Session.java]: I forgot to remove the period, my mistake

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java (hunk 3 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:619]: I did not reach this method yet, it was the formatter that remove this line. But I consider your comment when i reach this method

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java (hunk 2 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]: done

**Review by abenaru-zz — APPROVED** [review]

File: presto-main/src/main/java/io/prestosql/Session.java [hunk 4]
```diff
@@ -789,11 +789,9 @@ public ResourceEstimateBuilder setPeakMemory(DataSize peakMemory)
         }
 
         /**
-         * Estimate the resource usage for an executed query.
-         * <p>
-         * Takes the estimate of the CPU runtime and memory used for an executed query.
+         * Constructs the object that contains the estimated resource usage of a query.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/Session.java]:
```suggestion
         * Builds the object responsible for estimating the resource usage of a query (CPU and memory).
```

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 5]
```diff
@@ -209,13 +209,19 @@ public static Identity extractAuthorizedIdentity(Optional<Identity> optionalAuth
     }
 
     /**
-     * Creates a session identity.
-     *
-     * @param authenticatedIdentity an Optional Session {@link Identity}
-     * @param headers a MultivaluedMap containing all request headers.
-     * @param groupProvider a {@link GroupProvider} object.
+     * Extracts information from request headers about the user executing a request and his roles and credentials.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]:
```suggestion
     * Extracts information from request headers about the user requester and his roles and credentials.
```

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 6]
```diff
@@ -209,13 +209,19 @@ public static Identity extractAuthorizedIdentity(Optional<Identity> optionalAuth
...
+     * <p>
+     * The headers used to retrieve information are:
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]:
```suggestion
     * The headers used to retrieve this information are:
```

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 7]
```diff
@@ -209,13 +209,19 @@ public static Identity extractAuthorizedIdentity(Optional<Identity> optionalAuth
...
+     * - X-Presto-User
+     * - X-Presto-Extra-Credential
+     * - X-Presto-Role
      *
-     * @return a session identity object.
+     * @param authenticatedIdentity the authenticated user passed in the request
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]:
```suggestion
     * @param authenticatedIdentity the authenticated user defined in the request
```

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 8]
```diff
@@ -209,13 +209,19 @@ public static Identity extractAuthorizedIdentity(Optional<Identity> optionalAuth
...
+     * @param headers a map with all passed HTTP headers and respective values
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]:
```suggestion
     * @param headers a map with all defined HTTP headers and respective values
```

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 9]
```diff
@@ -456,14 +461,13 @@ public boolean supportClientTransaction()
     }
 
     /**
-     * Extracts the header PRESTO_CLIENT_TAGS.
+     * Extracts all tags passed on X-Presto-Client-Tags header.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]:
```suggestion
     * Extracts all tags defined on X-Presto-Client-Tags header.
```

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 10]
```diff
@@ -456,14 +461,13 @@ public boolean supportClientTransaction()
...
      * <p>
-     * Extract the header and add it to a Set.
-     * Returns a Set with the extracted {@code PRESTO_CLIENT_TAGS} header.
-     *
-     * @param headers headers a MultivaluedMap containing all request headers.
+     * The tags are passed as a comma-separated list of values and they are used to
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]:
```suggestion
     * The tags are defined as a comma-separated list of values and they are used to
```

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 11]
```diff
@@ -472,14 +476,13 @@ public boolean supportClientTransaction()
     }
 
     /**
-     * Extracts the header PRESTO_CLIENT_CAPABILITIES.
+     * Extracts all values passed in the X-Presto-Client-Capabilities header.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]:
```suggestion
     * Extracts all values defined in the X-Presto-Client-Capabilities header.
```

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 12]
```diff
@@ -472,14 +476,13 @@ public boolean supportClientTransaction()
...
      * <p>
-     * Extract the header and add it to a Set.
-     * Returns a Set with the extracted {@code PRESTO_CLIENT_CAPABILITIES} header.
+     * The values are passed as a comma-separated list and they are used when it is necessary to specify
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]:
```suggestion
     * The values are defined as a comma-separated list and they are used when it is necessary to specify
```

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 13]
```diff
@@ -472,14 +476,13 @@ public boolean supportClientTransaction()
...
+     * which are the functions that client support.
      *
-     * @param headers headers a MultivaluedMap containing all request headers.
-     *
-     * @return a Set with {@code PRESTO_CLIENT_CAPABILITIES} headers.
+     * @param headers a map with all passed HTTP headers and respective values
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]:
```suggestion
     * @param headers a map with all defined HTTP headers and respective values
```

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 14]
```diff
@@ -456,14 +461,13 @@ public boolean supportClientTransaction()
...
      * <p>
-     * Extract the header and add it to a Set.
-     * Returns a Set with the extracted {@code PRESTO_CLIENT_TAGS} header.
-     *
-     * @param headers headers a MultivaluedMap containing all request headers.
+     * The tags are passed as a comma-separated list of values and they are used to
+     * identify resource groups.
      *
-     * @return a Set with {@code PRESTO_CLIENT_TAGS} headers.
+     * @param headers a map with all passed HTTP headers and respective values
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]:
```suggestion
     * @param headers a map with all defined HTTP headers and respective values
```

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 15]
```diff
@@ -488,14 +491,16 @@ public boolean supportClientTransaction()
...
-     * If the value is not valid, an IllegalArgumentException is generated an handled.
-     *
-     * @param headers headers a MultivaluedMap containing all request headers.
+     * The currently estimated resources are:
+     * - execution time
+     * - cpu time
+     * - peak memory
      *
-     * @return {@link ResourceEstimates} that estimates resource usage for a query.
+     * @param headers a map with all passed HTTP headers and respective values
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]:
```suggestion
     * @param headers a map with all defined HTTP headers and respective values
```

**Review by jpedroantunes — CHANGES_REQUESTED** [review]

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java (hunk 5 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]: done

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java (hunk 6 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]: done

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java (hunk 7 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]: done

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java (hunk 8 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]: done

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java (hunk 9 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]: done

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java (hunk 10 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]: done

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java (hunk 11 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]: done

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java (hunk 12 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]: done

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java (hunk 13 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]: done

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java (hunk 14 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java]: done

**Review by jpedroantunes — APPROVED** [review]
